### PR TITLE
[FIX]mrp: workorder list view fixes

### PR DESCRIPTION
**Pull request by Dotni**
- changed the "start", "pause" and "done" button text to icons to be able to set the button column size at a fixed value
- moved the unblock button in the dropdown
- made the previous block button toggle between block and unblock
- change the dropdown bullet to red when blocked

### CODE REVIEW

File: addons/mrp/views/mrp_workorder_views.xml [hunk 1]
```diff
@@ -366,7 +364,7 @@
                             <div class="o_kanban_workorder_date h5">
                                 <field name="date_start"/>
                                 <field t-if="!record.date_start.raw_value" name="production_date"/>
-                            </div>
+                            </div>action_open_wizard
```
Comment (aab-odoo) [inline addons/mrp/views/mrp_workorder_views.xml:367]: oups

**Review by aab-odoo — APPROVED** [review]

File: addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js [hunk 2]
```diff
@@ -33,11 +35,25 @@ export class MOListViewDropdown extends Component {
     }
 
     get statusColor() {
+        if (this.isBlocked) {
+            return this.colorIcons[this.props.record.data.working_state];
+        }
         return this.colorIcons[this.state.state] || "";
```
Comment (svs-odoo) [inline addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js:41]: Don't you need empty string as fallback here too ?
```suggestion
        const state = this.isBlocked ? this.props.record.data.working_state : this.state.state;
        return this.colorIcons[state] || "";
```

File: addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js [hunk 3]
```diff
@@ -33,11 +35,25 @@ export class MOListViewDropdown extends Component {
...
+        return this.props.record.data.working_state == "blocked"
+    }
+
+    get blockTitle(){
+        if (this.isBlocked) {
+            return _t("Unblock");
+        }
+        return _t("Block");
+    }
+    async toggleBlock() {
```
Comment (svs-odoo) [inline addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js:54]:
```suggestion

    async toggleBlock() {
```

File: addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js [hunk 4]
```diff
@@ -33,11 +35,25 @@ export class MOListViewDropdown extends Component {
...
+        if (this.isBlocked) {
+            return this.colorIcons[this.props.record.data.working_state];
+        }
         return this.colorIcons[this.state.state] || "";
     }
 
-    async block() {
-        if (this.props.record.data.working_state == "blocked") {
+    get isBlocked() {
+        return this.props.record.data.working_state == "blocked"
```
Comment (svs-odoo) [inline addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js:45]:
```suggestion
        return this.props.record.data.working_state == "blocked";
```

File: addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js [hunk 5]
```diff
@@ -71,9 +80,20 @@ export class MOListViewDropdown extends Component {
             user.context
         );
     }
+
+    async callOrm(functionName){
+        let ids = this.props.record.model.root.selection?.map((element) => element.evalContext.id);
+        // if no records selected, take the current clicked one
+        if (!ids || (ids && ids.length == 0)) {
```
Comment (svs-odoo) [inline addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js:87]: if `!ids` is false, `ids` is then true
```suggestion
        if (!ids || ids.length == 0) {
```

**Review by svs-odoo — APPROVED** [review]

File: addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js [hunk 6]
```diff
@@ -5,9 +5,12 @@ import { DropdownItem } from "@web/core/dropdown/dropdown_item";
 import { registry } from "@web/core/registry";
 import { rpc } from "@web/core/network/rpc";
 import { standardWidgetProps } from "@web/views/widgets/standard_widget_props";
+import { _t } from "@web/core/l10n/translation";
 import { user } from "@web/core/user";
 import { useService } from "@web/core/utils/hooks";
 
+const workcenter_status_blocked = "blocked";
```
Comment (aab-odoo) [inline addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.js:12]: the convention is to camelcase variables in js

**Review by aab-odoo — APPROVED** [review]

**Comment by svs-odoo** [comment]
robodoo r+

File: addons/mrp/i18n/mrp.pot [hunk 7]
```diff
@@ -1097,7 +1097,13 @@ msgstr ""
 #. module: mrp
 #. odoo-javascript
 #: code:addons/mrp/static/src/components/wo_list_view_dropdown/wo_list_view_dropdown.xml:0
-msgid "Block WC"
+msgid "Block"
+msgstr ""
```
Comment (ticodoo) [inline addons/mrp/i18n/mrp.pot:1101]: :skull: don't manually change stuff in the pot file in the future please. Export the pot file + compare the diff. Duplicated terms like this break the Transifex sync + the terms are unicode alphabetized

**Pull request closed without merging**